.github/workflows/ci.yml

```diff
@@ -228,15 +228,58 @@ jobs:
           cd c_examples
           make
 
-  publish:
+  release:
+    # create a Github release
+    # only when
+    # - push on master
+    # - tag starts with 'v*'
     needs: [format, dummy, xen, kvm, virtualbox, hyper-v, c_api]
     runs-on: ubuntu-20.04
+    # output these value to be used by other jobs so they can add assets
+    outputs:
+      upload_url: ${{ steps.step_upload_url.outputs.upload_url }}
+      version: ${{ steps.get_version.outputs.version }}
+
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+
+      - name: install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+
+      - name: install cargo changelog
+        run: cargo install changelog
+
+      - name: generate changelog
+        run: changelog -o changelog.md
+
+      - name: Create a Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ steps.get_version.outputs.version }}
+          release_name: ${{ steps.get_version.outputs.version }}
+          body_path: changelog.md
+
+      - id: step_upload_url
+        run: echo "::set-output name=upload_url::${{ steps.create_release.outputs.upload_url }}"
```

Comment:
This step us unnecessary. Why not simply use `steps.create_release.outputs.upload_url` directly?

Reply:
The goal is to transfer the output_url from this job to other jobs would might want to add asset.
this will output the output_url in the `upload_url` parameter and be made accessible to the release_debian job

> Why not simply use steps.create_release.outputs.upload_url directly?

Could you add an example of your suggestion ?

Reply:
Well, afaik there is no difference between your `step_upload_url` variable and your `create_release` variable. If `step_upload_url` is available from other jobs, `create_release` should be too.